File: tests/fixtures/call-trace/nested.js
// Chain: a -> b -> c
function a() {
    b();
}

function b() {
    c();
}

function c() {
    console.log("leaf");
}

// Cycle: cycleA -> cycleB -> cycleA
function cycleA() {
    cycleB();
}

function cycleB() {
    cycleA();
}

// Multiple callers: caller1 -> target, caller2 -> target
function target() {
    console.log("target");
}

function caller1() {
    target();
}

function caller2() {
    target();
}

File: tests/fixtures/call-trace/nested.rb
# Chain: rb_a -> rb_b -> rb_c
def rb_a
  rb_b()
end

def rb_b
  rb_c()
end

def rb_c
  puts "leaf"
end

# Cycle: rb_cycle_a -> rb_cycle_b -> rb_cycle_a
def rb_cycle_a
  rb_cycle_b()
end

def rb_cycle_b
  rb_cycle_a()
end

# Multiple callers
def rb_target
  puts "target"
end

def rb_caller1
  rb_target()
end

def rb_caller2
  rb_target()
end
